### DISCUSSION TITLE
Adding a banner / warning in the generated files (like .md, .html, ...)

### DISCUSSION THREAD

**Discussion by cavo789** [quarto-dev/quarto-cli#8112] [Q&A]
### Description

Hi all

Did you know if it's possible to inject a warning at the top of the generated file like `// Please don't update this file because it's a generated one. Please update the .qmd file then render it again as .md`?

The idea is thus to notify the user to not make changes in a file generated by Quarto but to inform him that he need update the source file and render it again.

Thanks !

**Comment by mcanouil** [comment]
You can use a Lua filter:

````qmd
---
title: "Quarto Playground"
format: gfm
filters:
  - add_comment.lua
---

This is a playground for Quarto.
````

```lua
-- file: add_comment.lua
function Pandoc(doc)
  local comment = pandoc.RawBlock('html', '<!--- Generated file. Do not edit directly. --->')
  table.insert(doc.blocks, 1, comment)
  return doc
end
```

Note that a comment in Quarto/Pandoc is tricky, depending on what it is for exactly, see:
- https://github.com/quarto-dev/quarto-cli/discussions/3330

**Reply by cavo789** [reply]
Merci Michaël ! J'apprécie fortement !

Thanks Michaël, I really appreciate. I'll try this next week (first learn what is a filter and how implement it).

**Comment by cavo789** [comment]
**Once again, thank you @mcanouil  !**

I'm not sure I'm doing it right.

I've followed the [Quick Start](https://quarto.org/docs/extensions/filters.html#quick-start) documentation for creating a new filter and got this structure:

```bash
❯ tree
.
├── add_banner
│   ├── README.md
│   ├── _extensions
│   │   └── add_banner
│   │       ├── _extension.yml
│   │       └── add_banner.lua
│   └── example.qmd
└── index.qmd
```

I've copied your LUA code in file `add_banner/_extensions/add_banner/add_banner.lua` and your markdown example in `index.qmd`.

Running `quarto render index.qmd` fails with this error: `ERROR (/opt/quarto/share/filters/main.lua:3440) cannot open add_banner.lua: No such file or directory`.

I understand from the official doc I need to have a folder `_extensions` (and not `add_banner/_extensions`) so I move folder `add_banner/_extensions` one level up (and remove the `add_banner` folder):

```bash
❯ tree
.
├── _extensions
│   └── add_banner
│       ├── _extension.yml
│       └── add_banner.lua
├── index.md
└── index.qmd
```

The last thing I need to do to make it working is to replace in my `index.qmd` file the line `- add_comment.lua` and remove the extension so replace with `- add_comment`.

**And now, it works as expected.**

Once again, many thanks for the great support.
